roles/network_plugin/contiv/templates/contiv-etcd.env.j2

```diff
@@ -0,0 +1,23 @@
+# contiv etcd config
+{% if inventory_hostname in groups['kube-master'] %}
+export ETCD_DATA_DIR=/var/lib/etcd/contiv-data
+{% set listen_ip = ip | default('0.0.0.0') %}
+export ETCD_ADVERTISE_CLIENT_URLS=http://{{ listen_ip }}:{{ contiv_etcd_listen_port }}
+export ETCD_INITIAL_ADVERTISE_PEER_URLS=http://{{ listen_ip }}:{{ contiv_etcd_peer_port }}
+export ETCD_LISTEN_PEER_URLS=http://{{ listen_ip }}:{{ contiv_etcd_peer_port }}
+export ETCD_LISTEN_CLIENT_URLS=http://{{ listen_ip }}:{{ contiv_etcd_listen_port }},http://127.0.0.1:{{ contiv_etcd_listen_port }}
```

Comment:
This line is what broke my env. You can't bind to 0.0.0.0 and then 127.0.0.1. Let's change listen_ip to

```
{% set listen_ip = ip | default(ansible_default_ipv4['address']) %}
```
Additionally, you should set up this logic for listen/bind address in roles/contiv/defaults/main.yml or as set_fact in a task. We should keep the logic embedded in a jinja2 template as minimal as possible whenever possible. I know we don't do it everywhere, but we already do it for etcd.

Reply:
I'll add it in next commit, and @unclejack will cherry pick it.
Although personally i think set_fact in task may not be as readable as in jinja2 template.